### PR TITLE
form components proactive help

### PR DESCRIPTION
**Pull request by timisenco2015**
# Description

This feature allows the form designers to view help details or component information for each of the form.io components.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/frontend/src/components/admin/AdminPage.vue [hunk 1]
```diff
@@ -1,12 +1,14 @@
 <template>
   <v-tabs>
-    <v-tab>Forms</v-tab>
+    <v-tab >Formss</v-tab>
```
Comment (jasonchung1871) [inline app/frontend/src/components/admin/AdminPage.vue:3]: typo here

File: app/frontend/src/components/infolinks/GeneralLayout.vue [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  },
+
+  props: {
+    layoutList: {
+      type:Array,
+      required: true
+    },
+    componentsList: {
+      type:Array,
+      defualt:[]
```
Comment (jasonchung1871) [inline app/frontend/src/components/infolinks/GeneralLayout.vue:101]: typo? should it be default?

File: app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+
+  it('isComponentPublish()', async () => {
+    const wrapper = shallowMount(GeneralLayout, {
+      localVue,
+      stubs: ['InformationLinkDialog','InformationLinkPreviewDialog'],
+      propsData: {
+        componentsList: {
+          type:Array,
+          defualt:[]
```
Comment (jasonchung1871) [inline app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js:18]: typo? should it be default?

File: app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const getComponentSpy = jest.spyOn(GeneralLayout.methods, 'getComponent');
+    const onDialogSpy = jest.spyOn(GeneralLayout.methods, 'onDialog');
+
+    const wrapper = shallowMount(GeneralLayout, {
+      localVue,
+      stubs: ['InformationLinkDialog','InformationLinkPreviewDialog'],
+      propsData: {
+        componentsList: {
+          type:Array,
+          defualt:[]
```
Comment (jasonchung1871) [inline app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js:46]: typo? should it be default?

File: app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+  it('onPreviewDialog()', async () => {
+
+    const wrapper = shallowMount(GeneralLayout, {
+      localVue,
+      stubs: ['InformationLinkDialog','InformationLinkPreviewDialog'],
+      propsData: {
+        componentsList: {
+          type:Array,
+          defualt:[]
```
Comment (jasonchung1871) [inline app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js:70]: typo? should it be default?

File: app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+  it('onDialog()', async () => {
+
+    const wrapper = shallowMount(GeneralLayout, {
+      localVue,
+      stubs: ['InformationLinkDialog','InformationLinkPreviewDialog'],
+      propsData: {
+        componentsList: {
+          type:Array,
+          defualt:[]
```
Comment (jasonchung1871) [inline app/frontend/tests/unit/components/infolinks/GeneralLayout.spec.js:93]: typo? should it be default?

**Comment by jasonchung1871** [comment]
make a search for `textfiled`, the typo shows up a lot. this should be `textfield`.

**Comment by timisenco2015** [comment]
Persistent "column \"imagename\" of relation \"form_components_proactive_help\" does not exist" error message. I will need to create new and fresh PR

**Pull request closed without merging**